vector3d.h

```diff
@@ -0,0 +1,83 @@
+//3d vector class for Rendera, Yashas M Salian
+
+#include<math.h>
+
+class vec3
+{
+private:
+    float x, y, z;    //Components of vector are kept private
```

Comment:
A nice enhancement would be to make it so that the user can easily switch between `float` and `double` vectors. Maybe a templated `vec3` class or a preprocessor macro?